zk_ee/src/system/mod.rs

```diff
@@ -100,6 +101,10 @@ impl<S: SystemTypes> System<S> {
         }
     }
 
+    pub fn get_interop_roots(&self) -> [InteropRoot; 100] {
```

Comment:
Again, why 100? Should it be just `&[InteropRoot]`?

Reply:
nit, bur agree with Vlad that it's better to use slice here, otherwise we'll copy interoop roots with this functions calls.

As for borrowing issues, I think you can just rewrite loop in the bootloader, to borrow interop roots on each iteration only for calldata preparation. Alternatively, you can make this method return Option, and use `take` for return, but slice is better IMHO.